From: Michael Ellerman <michael@ellerman.id.au>
Date: Wed, 25 Jan 2006 21:48:48 +1300
Subject: [PATCH] powerpc: Refuse to boot a kdump kernel via OF

You can't boot a kdump kernel via OF, not reliably anyway, the kernel being at
32 MB conflicts with the zImage wrapper etc. and it blows up.

It's trivial to check in prom_init though, and this is early enough that we can
actually drop back to OF where a reset-all will get you going again, which is
kinda nice. I think this should go in for 2.6.16.

Signed-off-by: Olaf Hering <olh@suse.de>
 arch/powerpc/kernel/prom_init.c |    4 ++++
 1 files changed, 4 insertions(+)

Index: linux/arch/powerpc/kernel/prom_init.c
===================================================================
--- linux.orig/arch/powerpc/kernel/prom_init.c
+++ linux/arch/powerpc/kernel/prom_init.c
@@ -2098,6 +2098,10 @@ unsigned long __init prom_init(unsigned 
 	 */
 	prom_init_stdout();
 
+	/* Bail if this is a kdump kernel. */
+	if (PHYSICAL_START > 0)
+		prom_panic("Error: You can't boot a kdump kernel from OF!\n");
+
 	/*
 	 * Check for an initrd
 	 */
_______________________________________________
Linuxppc64-dev mailing list
Linuxppc64-dev@ozlabs.org
https://ozlabs.org/mailman/listinfo/linuxppc64-dev

